### PR TITLE
feat: proof of work audit part 4

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
- Added documentation to public methods
- Renamed the Monero PoW algorithm to its proper name, RandomX

Motivation and Context
---
Preparation for code audit

How Has This Been Tested?
---
No functional changes were made, compiling successfully with clippy,

What process can a PR reviewer use to test or verify this change?
---
Review changes

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: base_layer/core/src/blocks/accumulated_data.rs [hunk 1]
```diff
@@ -284,16 +284,16 @@ impl BlockHeaderAccumulatedDataBuilder<'_> {
             field: "Current achieved difficulty",
         })?;
 
-        let (monero_diff, blake_diff) = match achieved_target.pow_algo() {
```
Comment (stringhandler) [inline base_layer/core/src/blocks/accumulated_data.rs:287]: blake_diff should be renamed here as well

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 2]
```diff
@@ -79,7 +79,7 @@ pub struct ConsensusConstants {
     pub(in crate::consensus) emission_decay: &'static [u64],
     /// This is the emission curve tail amount
     pub(in crate::consensus) emission_tail: MicroTari,
-    /// This is the maximum age a monero merge mined seed can be reused
+    /// This is the maximum age a RandomX merge mined seed can be reused
```
Comment (stringhandler) [inline base_layer/core/src/consensus/consensus_constants.rs]: I think this is specific enough to Monero to remain Monero

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 3]
```diff
@@ -280,7 +280,7 @@ impl ConsensusConstants {
         }
     }
 
-    /// The maximum age a monero merge mined seed can be reused
+    /// The maximum age a RandomX merge mined seed can be reused
```
Comment (stringhandler) [inline base_layer/core/src/consensus/consensus_constants.rs]: same

**Review by stringhandler — DISMISSED** [review]
Approve for the most part, but some comments

File: applications/tari_app_grpc/proto/base_node.proto [hunk 4]
```diff
@@ -190,7 +190,7 @@ message NetworkDifficultyResponse {
     uint64 timestamp = 4;
     uint64 pow_algo = 5;
     uint64 sha3_estimated_hash_rate = 6;
```
Comment (SWvheerden) [inline applications/tari_app_grpc/proto/base_node.proto]: Can you change SHA3 -> SHA3x

File: base_layer/core/src/proof_of_work/sha3x_pow.rs [hunk 5]
```diff
@@ -37,7 +37,8 @@ pub fn sha3x_difficulty(header: &BlockHeader) -> Result<Difficulty, DifficultyEr
     Ok(sha3x_difficulty_with_hash(header)?.0)
 }
 
-pub fn sha3_hash(header: &BlockHeader) -> Vec<u8> {
+/// Calculate the Tari Sha3 mining hash
+pub fn sha3x_hash(header: &BlockHeader) -> Vec<u8> {
```
Comment (SWvheerden) [inline base_layer/core/src/proof_of_work/sha3x_pow.rs]: leave this sha3

File: base_layer/core/src/proto/block.proto [hunk 6]
```diff
@@ -12,7 +12,7 @@ package tari.core;
 
 // Metadata required for validating the Proof of Work calculation
 message ProofOfWork {
-    // 0 = Monero
+    // 0 = RandomX
     // 1 = Blake
```
Comment (SWvheerden) [inline base_layer/core/src/proto/block.proto]: sha3x

File: base_layer/core/src/proto/block.proto [hunk 7]
```diff
@@ -84,7 +84,7 @@ message HistoricalBlock {
 
 message BlockHeaderAccumulatedData {
     uint64 achieved_difficulty = 1;
-    uint64 accumulated_monero_difficulty = 2;
+    uint64 accumulated_randomx_difficulty = 2;
     uint64 accumulated_sha_difficulty = 3;
```
Comment (SWvheerden) [inline base_layer/core/src/proto/block.proto]: sha3x

File: applications/tari_app_grpc/proto/block.proto [hunk 8]
```diff
@@ -66,19 +66,19 @@ message BlockHeader {
 // Metadata required for validating the Proof of Work calculation
 message ProofOfWork {
     // 0 = Monero
-    // 1 = Blake
+    // 1 = Sha3X
     uint64 pow_algo = 1;
-//    uint64 accumulated_monero_difficulty = 2;
-//    uint64 accumulated_sha_difficulty = 3;
+//    uint64 accumulated_randomx_difficulty = 2;
+//    uint64 accumulated_sha3x_difficulty = 3;
```
Comment (SWvheerden) [inline applications/tari_app_grpc/proto/block.proto]:
```suggestion
```

File: applications/tari_app_grpc/proto/block.proto [hunk 9]
```diff
@@ -66,19 +66,19 @@ message BlockHeader {
...
     bytes pow_data = 4;
    // uint64 target_difficulty = 5;
```
Comment (SWvheerden) [inline applications/tari_app_grpc/proto/block.proto]:
```suggestion
```

File: applications/tari_app_grpc/proto/block.proto (hunk 8 above)
Comment (hansieodendaal) [reply applications/tari_app_grpc/proto/block.proto]: Addressed in #5521

File: applications/tari_app_grpc/proto/block.proto (hunk 9 above)
Comment (hansieodendaal) [reply applications/tari_app_grpc/proto/block.proto]: Addressed in #5521

**Pull request merged**